File: implement-shell-tools/wc/wcfile.js
const fs = require("fs");

const args = process.argv.slice(2);
const files = args.filter((a) => !a.startsWith("-"));

files.forEach((file) => {
  const content = fs.readFileSync(file, "utf-8");

  const lines = content.split("\n").length;
  const words = content.trim().split(/\s+/).length;
  const chars = content.length;

  console.log(`${lines} ${words} ${chars} ${file}`);
});

File: implement-shell-tools/wc/wcfile.py
import sys

args = sys.argv[1:]

files = [a for a in args if not a.startswith("-")]

for file in files:
    with open(file, "r") as f:
        content = f.read()

        lines = len(content.split("\n"))

        words = len(content.strip().split())

        chars = len(content)

        print(f"{lines} {words} {chars} {file}")